### PR TITLE
EventEmitter emit spec update

### PR DESCRIPTION
**Pull request by mattheworiordan**
Clarify expected behaviour when emit is called and existing listeners mutate the registered listeners.

See https://github.com/ably/ably-js/issues/364 and implementation in Ruby at https://github.com/ably/ably-ruby/pull/99/commits/bd5495df50b82fa6c9152aaf15ccd671588afea1

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Note this is a PR to be merged into 0.10

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -587,6 +587,7 @@ h3(#eventemitter). EventEmitter mixin / interface
...
 * @(RTE6)@ @EventEmitter#emit@ emits an event, calling registered listeners with the given event name and any other given arguments. If an exception is raised in any of the listeners, the exception is caught by the @EventEmitter@ and the exception is logged to the Ably logger. Tests must exist to ensure exceptions raised in client code do not propagate and inhibit other event processing within the client library
+** @(RTE6b)@ Every listener that is registered at the time @emit@ is invoked should be called exactly once. A test should assert that any listener registered and/or removed within an already existing @on@ listener does not result in the newly added listeners being called or the newly removed listeners not being called when an event is emitted, see "https://goo.gl/OVTtjO":https://goo.gl/OVTtjO
```
Comment (SimonWoolf) [inline content/client-lib-development-guide/features.textile]: > RTE6b

Why not RTE6a?

> `Every listener that is registered at the time @emit@ is invoked should be called exactly once.`

This seems like an odd way of putting the requirement. E.g. if I add an onMessage listener within an onMessage callback, I expect it to be called maybe hundreds of times, whenever a message arrives, not 'exactly once'. I just don't want it to be called on the current invocation of `emit`.

Alternative wording proposal: https://github.com/ably/docs/compare/source-0-10...event-emitter-emit-spec-update-alternative?expand=1

Comment (mattheworiordan) [inline content/client-lib-development-guide/features.textile]: > Why not RTE6a?

Typo!

> Alternative wording proposal:

I love it and have used your wording now with one spelling error fix

**Comment by SimonWoolf** [comment]
:+1:

**Comment by paddybyers** [comment]
LGTM

**Pull request merged**